xl-v1.go

```diff
@@ -435,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
+	xl.lockNS(volume, path, readLock)
+	_, metadata, doSelfHeal, err := xl.getReadableDisks(volume, path)
+	xl.unlockNS(volume, path, readLock)
+	if err != nil {
+		return FileInfo{}, err
+	}
+
+	if doSelfHeal {
+		if err = xl.selfHeal(volume, path); err != nil {
```

Comment:
if selfHeal() is not a goroutine or not using goroutine inside, there is chance of time out error to the client.  Is it expected now?